### PR TITLE
Upgrade blueprint workflow

### PR DESCRIPTION
**Pull request by pitmonticone**
- [x] Add blueprint-relevant path filters to avoid unnecessary action runs
- [x] Cancel previous runs if a new commit is pushed to the same PR or branch.

### CODE REVIEW

File: .github/workflows/blueprint.yml [hunk 1]
```diff
@@ -3,14 +3,38 @@ name: Compile blueprint
 on:
   push:
     branches:
-      - main # Trigger on pushes to the default branch
-  workflow_dispatch:        # Allow manual triggering of the workflow from the GitHub Actions interface
-
-# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+      - main
+    paths:
```
Comment (mo271) [inline .github/workflows/blueprint.yml:7]: not sure if this is really needed, also it is not capturing all relevant files in my repo, at least I see that AUTHORS.md is missing, but this is used when building the latex.

Comment (pitmonticone) [reply .github/workflows/blueprint.yml:7]: Yes, that is a minimal set of path filters. You can add all those which are project-specific.

**Comment by mo271** [comment]
This is still failing to build, although it seems like the permissions are ok. Perhaps we can test each commit here separately?

**Comment by pitmonticone** [comment]
This one failed too. It's definitely not a problem relative to any change.

**Comment by mo271** [comment]
Ok, two summarize two changes:
 - add `if` to deploy step
 -  use paths-ignore to have a shorter list of files (also to err on the safe side adding new files).

**Pull request merged**